### ISSUE TITLE
[dotnet-sdk-8.0.100-alpha.1.22619.1] When multiple toolstrips add to top panel of ToolStripContainer, System.InvalidCastException will throw

### ISSUE THREAD

**Issue by Junjun-zhao** [dotnet/winforms#8415]
### .NET version

```
.NET SDK:
Version:   8.0.100-alpha.1.22619.1
Commit:    8750a90545
Runtime Environment:
OS Name:     Windows
OS Version:  10.0.19045
OS Platform: Windows
RID:         win10-x64
Base Path:   C:\Program Files\dotnet\sdk\8.0.100-alpha.1.22619.1\

Host:
 Version:      8.0.0-alpha.1.22605.1
 Architecture: x64
 Commit:       1a37caf773

.NET SDKs installed:
  8.0.100-alpha.1.22619.1 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 8.0.0-alpha.1.22615.1 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 8.0.0-alpha.1.22605.1 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 8.0.0-alpha.1.22614.1 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
```

### Did it work in .NET Framework?

Yes

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

**Verify Scenarios:**
1). Windows10 21H2 x64+dotnet-sdk-8.0.100-alpha.1.22621.5: Fail
2). Windows10 21H2 x64+dotnet-sdk-8.0.100-alpha.1.22619.1: Fail
3). Windows10 21H2 x64+dotnet-sdk-8.0.100-alpha.1.22616.7: Fail
4). Windows 10 21H2 x64 + dotnet-sdk-7.0.101: Pass

### Issue description

When testing the 3rd Party App with .NET 8 runtime,  System.InvalidCastException will throw if multiple toolstrips are added to top panel of ToolStripContainer.

**Application Name**:  mRemoteNG
**OS**: Windows 10 21H2
**CPU:** X64
**.NET Build Number:** dotnet-sdk-8.0.100-alpha.1.22619.1

App or App Source code checking at:  https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1709163
**Github Link:** 
https://github.com/mRemoteNG/mRemoteNG

### Steps to reproduce

**Repro steps:**
**The machine only has 8.0 installed. And DOTNET_ROLL_FORWARD_ON_NO_CANDIDATE_FX=2 is set.**

1. Copy mRemoteNG app to local machine.
2. Launch mRemoteNG.exe

**Expected Result:** 
Application can launch successfully.

**Actual Result:**
Application launch failed with below exception:
![image](https://user-images.githubusercontent.com/83067197/209086429-9bec6a8c-aa51-408c-af8a-1f24f6851da5.png)

**Minimal Repro sample:**
Minial repro sample attached.[ToolTipDemo.zip](https://github.com/dotnet/winforms/files/10284921/ToolTipDemo.zip)

1. Create a .NET 7.0 WinForm app.
2. Add following code in Form1.Designer.cs file:
```
       private ToolStripContainer toolStripContainer1;
        private ToolStrip toolStrip1;
        private ToolStrip toolStrip2;

        private void InitializeComponent()
        {
            this.components = new System.ComponentModel.Container();
            this.AutoScaleMode = System.Windows.Forms.AutoScaleMode.Font;
            this.ClientSize = new System.Drawing.Size(800, 450);
            this.Text = "Form1";

            toolStripContainer1 = new ToolStripContainer();
            toolStrip1 = new ToolStrip();
            toolStrip2 = new ToolStrip();
            toolStrip1.Items.Add("One");
            toolStrip2.Items.Add("Two");
            toolStripContainer1.TopToolStripPanel.Controls.Add(toolStrip1);
            //we add second to repro, if we remove this line, it will work
            toolStripContainer1.TopToolStripPanel.Controls.Add(toolStrip2);
            Controls.Add(toolStripContainer1);
        }
```
5. Build the project.
6. Change the runtimeconfig to let the app run against with .NET 8.0.100-alpha.1.22619.1:
```
{
  "runtimeOptions": {
    "tfm": "net7.0",
    "frameworks": [
      {
        "name": "Microsoft.NETCore.App",
        "version": "8.0.0-alpha.1.22605.1"
      },
      {
        "name": "Microsoft.WindowsDesktop.App",
        "version": "8.0.0-alpha.1.22614.1"
      }
    ]
  }
}
```
8. Press F5 to launch the sample app.

**Excepted Result:**
App can launch and toolstrips will load on the form.

**Actual Result:**
This exception is thrown in code:
```
System.InvalidCastException: 'Unable to cast object of type 'System.Collections.Generic.List`1[System.Windows.Forms.Layout.IArrangedElement]' to type 'System.Collections.Generic.IEnumerable`1[System.Windows.Forms.ToolStripPanelCell]'.'
```
**Findings:**
When adding multiple tooltstrips to top panel of ToolStripContainer will cause this issue. If only 1 tooltip added, it won't trigger this exception.

@dotnet-actwx-bot @dotnet/compat

**Comment by gpetrou** [comment]
Looks like this was broken in https://github.com/dotnet/winforms/pull/8207. I will add a fix.

**Comment by Junjun-zhao** [comment]
Verified and it has been fixed on dotnet-sdk-8.0.100-alpha.1.23060.3 .
